### PR TITLE
Tests for full grid scan

### PR DESCRIPTION
**Pull request by Tom-Willemsen**
Fixes https://github.com/DiamondLightSource/python-artemis/issues/674

Adds a unit test for `full_grid_scan`

Note: depends on https://github.com/DiamondLightSource/python-artemis/pull/722

### CODE REVIEW

File: src/artemis/experiment_plans/tests/test_full_grid_scan_plan.py
Comment (DominicOram) [inline src/artemis/experiment_plans/tests/test_full_grid_scan_plan.py:114]: Could: A few type hints in here would be nice

**Review by DominicOram — CHANGES_REQUESTED** [review]
Generally good, minor tidying up I think you could do. Also some minor workflow issues:
* We usually to add the issue number in the branch name and commit messages e.g. see https://github.com/DiamondLightSource/python-artemis/pull/708. People are a bit hit and miss about it though, not a big deal.
* vscode should be sorting imports on save, do you have the following in your `settings.json`?
```
"editor.codeActionsOnSave": {
        "source.organizeImports": true
    },
    "python.sortImports.args": [
        "--profile",
        "black"
    ],
```

**Review by DominicOram — APPROVED** [review]
Looks good, wait until #722 is merged but otherwise fine

**Comment by dperl-dls** [comment]
merging as #722  is merged now

**Pull request merged**
